api/db-cdr/generate-cdr/libproject/genomic_manifests.rb

```diff
@@ -374,9 +382,55 @@ def _build_copy_manifest_row(
   }
 end
 
-def build_manifests_for_aw4_section(input_section, ingest_bucket, dest_bucket, display_version_id, aw4_rows, output_manifest_path)
-  # TODO(RW-8269): handle delta directories
-  path_prefix = "pooled/#{input_section["pooledDestPathInfix"]}/#{display_version_id}_base"
+# Logic to read the previous manifest for a delta release.
+# Returns a hash of person_id => CSV row found in previous manifest
+def _read_previous_manifest(project, dest_bucket, deltaReleaseManifestPath, deltaRelease, infix)
+  common = Common.new
+  prev_manifest = ""
+  manifest_path = deltaReleaseManifestPath
+
+  deploy_account = must_get_env_value(project, :publisher_account)
+
+  # If deltaReleaseManifestPath is specified, try to use it.
+  unless deltaReleaseManifestPath.nil?
+    if deltaReleaseManifestPath.start_with?("gs://")
+      prev_manifest = common.capture_stdout(["gsutil", "-i", deploy_account, "cat", deltaReleaseManifestPath])
+    else
+      prev_manifest = IO.read(deltaReleaseManifestPath)
+    end
+  end
+
+  # Try to find the manifest given the deltaRelease field.
+  unless deltaRelease.nil? or not prev_manifest.empty?
```

Comment:
I'm having trouble with all this negation... let me work through this a bit.

```
unless ( (release.nil?) || ! (manifest.empty?) )
=
if ( !(release.nil?) && manifest.empty? )
=
if (manifest.empty? && release exists)
```

ok I think the last iteration is clearer.  Would you mind making that change?

Reply:
makes sense, will change it